### PR TITLE
[fix](prepare) fix datetimev2 return err when binary_row_format

### PR DESCRIPTION
**Pull request by nextdreamblue**
## Proposed changes

Issue Number: close #xxx

1. fix datetimev2 return err when binary_row_format. before pr, Backend return datetimev2 alwary by to_string. 
2. fix datatimev2 return metadata loss scale.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run compile

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
sh buildall

**Comment by nextdreamblue** [comment]
run  buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java [hunk 1]
```diff
@@ -298,6 +298,8 @@ public int getMysqlDecimals(Type type) {
             case DECIMAL64:
             case DECIMAL128:
             case DECIMAL256:
+            case TIMEV2:
+            case DATETIMEV2:
```
Comment (eldenmoon) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java:302]: should not be in `getMysqlDecimals` for TIMEV2 and DATETIMEV2, function used for decimal

Comment (nextdreamblue) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSerializer.java:302]: mysql use this for datatime
image
image

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

**Review by cambyzju — DISMISSED** [review]
LGTM

**Comment by nextdreamblue** [comment]
run buildall

**Review by xy720 — APPROVED** [review]
LGTM

**Review by cambyzju — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**